docs/code-organization/git-quickstart.mdx

```diff
@@ -0,0 +1,331 @@
+	new file:   ExampleTestFile.txt
+```
+
+To unstage files (tell Git to forget about the current version that you added),
+use [`git rm --cached <file>`][git-rm] as suggested by the output provided in
+[`git status`][git-status].
+
+:::caution
+Note that `--cached` tells Git to only remove the cached version and not the
+file itself! Failing to specify `--cached` will cause the file to be deleted as
+well.
+:::
+
+```
+$ git rm --cached ExampleTestFile.txt
+rm 'ExampleTestFile.txt'
+
+$ git status
+On branch main
+
+No commits yet
+
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+	ExampleTestFile.txt
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+You may specify multiple files at once for both commands by separating them
+with spaces.
+```
+$ git add a.txt b.txt
+$ git rm --cached a.txt b.txt
+rm 'a.txt'
+rm 'b.txt'
+
+```
+
+## Creating Commits
+
+To add the current snapshot you crafted with [`git add`][git-add] into the
+actual history, use [`git commit`][git-commit].
+
+Please use `-m` to specify the commit message. The first `-m` will specify the
+title of the commit and the second `-m` will specify the body of the commit.
+
+```
+$ git commit -m "docs: Fix typo in Example docs" -m "Example should be spelled with a single e rather than two es.
+dquote>
+dquote>[SCSE-24]"
+[main 4faee54] docs: Fix typo in Example docs
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+```
+
+Note that in this example, `dquote>` should not be typed in. It simply means
+that the text after is still part of the double quotes.
+
+For an idea of what to write, please refer to our
+[conventions](naming-convention#commit-messages) document.
+
+### Fixing Mistakes
+
+No one gets their code correct every single time and mistakes happen. If you
+messed up the content of a commit (the changes you made) or the title/message,
+you can just re-create the commit with the `--amend` flag!
+
+This will replace the latest commit with the new one that hopefully is fine
+this time. ;)
+
+```
+$ git commit --amend -m "docs: Fix typo in Example docs" -m "Example should be spelled with a single e rather than two es.
+dquote>
+dquote>[SCSE-24]"
+[main f253a79] docs: Fix typo in Example docs
+ Date: Thu Oct 13 23:39:33 2022 +0800
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+```
+
+:::caution
+Note that by doing this, you are recreating history!
+NEVER ever re-create history that is already public and in the `main` branch as
+it will force everyone to change their history as well.
+:::
+
+## Branching
+
+You should definitely create a branch for what you are working on before you
+start working on your super cool feature.
+
+To quote [Pro Git Chapter 3][pro-git-chap-3], branching means you diverge from
+the main line of development and continue to do work without messing with
+that main line.
+
+The [`git switch`][git-switch] command has been added in a recent update of Git
```

Comment:
let's specify that this is only available git v2.23+, and the fact that it's experimental, with api subject to change.